libvast/src/system/index.cpp

```diff
@@ -393,18 +393,9 @@ caf::behavior index(caf::stateful_actor<index_state>* self, const path& dir,
     VAST_DEBUG(self, "received exit from", msg.source,
                "with reason:", msg.reason);
     auto& st = self->state;
-    if (!st.unpersisted.empty() || st.active_partition_indexers > 0) {
-      auto delay = defaults::index::shutdown_retry_interval;
-      VAST_INFO(self, "delaying exit by", delay,
-                "to wait for outstanding indexers due to",
-                st.unpersisted.size(), "unpersisted and ",
-                st.active_partition_indexers, "active");
-      // Tell the upstreams that we are stopping.
-      st.stage->shutdown();
-      st.stage->out().close();
-      self->delayed_send(self, delay, msg);
-      return;
-    }
+    // Tell the upstreams that we are stopping.
+    st.stage->shutdown();
+    st.stage->out().close();
```

Comment:
I think even this might not be necessary any more (RAII should work here). I'll test that.

Reply:
According to my tests it is save to remove.

Reply:
Removed.